### ISSUE TITLE
☂ appveyor tests are failing but appear to be succeeding

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8664]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)| @glyph reported|
|-|-|
|Trac ID|trac#8664|
|Type|defect|
|Created|2016-07-24 02:31:56Z|
|Branch|https://github.com/twisted/twisted/tree/8664-appveyor-enable|

The tests on appveyor are failing.

There are two kinds of test failures:

- [#8662](https://github.com/twisted/twisted/issues/8662)
- [#8663](https://github.com/twisted/twisted/issues/8663)

And this ticket should track enabling the status to fail when the test fails once those are fixed.

<details><summary>Searchable metadata</summary>

```
trac-id__8664 8664
type__defect defect
reporter__glyph glyph
priority__normal normal
milestone__None None
branch__8664_appveyor_enable 8664-appveyor-enable
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1469327516474787 1469327516474787
changetime__1469469290505177 1469469290505177
version__None None
owner__hawkowl hawkowl

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_1"></a>|@adiroiban commented|
|-|-|

ok... we now have OpenSSH installed on appveyor. I think we can add it to the cache directories so that we don't need to install it each time and only install if it is missing.

for my private project I am doing the following hack: packaging JS libs + other binaries (ex selenium ie/chrome drivers) as python packages with data files. In this way I can control their version and also cache them.... and is nice to manage them using pip :) ... not sure if this will make sense for Twisted

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_3"></a>|@hawkowl commented|
|-|-|

As per the other ticket, the cache dir probably won't get us much (it's a few mb so wouldnt really be worth it).

Removed the allowed failure and fixed the builds for apidocs so everything is green. Please review.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_4"></a>|@adiroiban set owner to @hawkowl|
|-|-|

OMG we haz green Windows build on Appveyor. Many thanks!

All good.

Please merge!

Feel free to remove the win7 build since it is not reporting coverage.
I took a look at the windows7 vm but I have no idea why the coverage publishing is denied.

Up next ... python 3.5 on Windows ? :)

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_5"></a>|@adiroiban set status to `closed`|
|-|-|

In changeset 3e4680864fd22802d2f21cb80c8e0f400991dc61
```
#!CommitTicketReference repository="" revision="3e4680864fd22802d2f21cb80c8e0f400991dc61"
Merge 8664-appveyor-enable: Have appveyor build report the actual test results.

Author: hawkowl
Reviewer: adiroiban
Fixes: #8664
```

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_6"></a>|@adiroiban commented|
|-|-|

ok.... before removing the win7 builders, we need to have the bdist job on Appveyor + iocp/select reactor jobs

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_7"></a>|@adiroiban commented|
|-|-|

Here are some follow up tickets

* windows bdist - [#8676](https://github.com/twisted/twisted/issues/8676)
* iocp / select test runs - [#8677](https://github.com/twisted/twisted/issues/8677)
* fix iocp tests on Appveyor - [#8679](https://github.com/twisted/twisted/issues/8679)
* py3.5 on windows on Appveyor - [#8678](https://github.com/twisted/twisted/issues/8678)
